### PR TITLE
feat: Implement PubsubLiteConsumer.

### PR DESCRIPTION
**Pull request by dpcollins-google**
This implements Kafka's Consumer API.

### CODE REVIEW

File: pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+import org.apache.kafka.clients.consumer.OffsetAndTimestamp;
+import org.apache.kafka.clients.consumer.OffsetCommitCallback;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+
+/**
+ * A class that uses a MinimalKafkaConsumer to remove the duplicate methods from the kafka consumer.
```
Comment (manuelmenzella-google) [inline pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java:65]: What is a MinimalKafkaConsumer? I cannot find references to it.

File: pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+  public ConsumerRecords<byte[], byte[]> poll(long l) {
+    return poll(Duration.ofMillis(l));
+  }
+
+  @Override
+  public ConsumerRecords<byte[], byte[]> poll(Duration timeout) {
+    return requireValidConsumer().poll(timeout);
+  }
+
+  Map<Partition, Offset> checkAndTransform(Map<TopicPartition, OffsetAndMetadata> map) {
```
Comment (manuelmenzella-google) [inline pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java:250]: I would personally make these method names more descriptive. "checkAndTransform" doesn't really convey what this method is doing without looking at the return/argument types, which are not always immediately visible at the call sites.

There's others like this: toKafka, toDuration, etc, but I think those are more obvious.

**Review by manuelmenzella-google — APPROVED** [review]
Minor nits. Thanks.

File: pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java (hunk 1 above)
Comment (dpcollins-google) [reply pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java]: Done.

File: pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java (hunk 2 above)
Comment (dpcollins-google) [reply pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumer.java]: Done.

**Pull request merged**